File: Android/app/src/main/java/com/caffeine/swingit/GameOverScreen.java
package com.caffeine.swingit;

import android.graphics.Typeface;

import com.caffeine.swingit.Graphics.GLabel;
import com.caffeine.swingit.Graphics.GPoint;
import com.caffeine.swingit.Graphics.GTools;

import java.util.List;
import java.util.Timer;
import java.util.TimerTask;

/**
 * Ecran de game over
 * Created by olivierpicard on 06/04/2018.
 */

public class GameOverScreen {
    private final int BUTTON_INTER_SPACE;
    private final GameScene scene;
    private Button retry_button;
    private Button menu_button;
    private boolean isHidden;


    public GameOverScreen(GameScene scene) {
        this.scene = scene;
        this.isHidden = true;
        BUTTON_INTER_SPACE = (int)this.scene.getSize().height / 15;

        this.retry_button = new Button(R.string.recommencer);
        this.retry_button.setPosition(GTools.fromSceneToScreenPos(this.scene.getSize(), new GPoint(0.5f, 0.6f)));

        this.menu_button = new Button(R.string.accueil);
        this.menu_button.setPosition(new GPoint(this.retry_button.getPosition().x,
                this.retry_button.getPosition().y + this.retry_button.getSize().height + BUTTON_INTER_SPACE));
    }


    public void hide() {
        if(isHidden) return;
        this.isHidden = true;
        scene.removeChild(this.retry_button);
        scene.removeChild(this.menu_button);
    }


    public void show() {
        if(!isHidden) return;
        this.isHidden = false;
        this.scene.addChild(this.retry_button);
        this.scene.addChild(this.menu_button);
    }


    public void touchUp(GPoint pos){
        if(retry_button.isClicked(pos))
            scene.setFlagGameState(GameScene.GameState.PLAY);
        else if(menu_button.isClicked(pos))
            scene.setFlagGameState(GameScene.GameState.WELCOME);
    }

}

File: iOS/Swing-It/Swing-It/GameOverScreen.swift
//
//  GameOverScreen.swift
//  Swing-It
//
//  Created by Olivier Picard on 03/10/2018.
//  Copyright © 2018 Caffeine. All rights reserved.
//

import Foundation
import SpriteKit

class GameOverScreen {
    let scene: GameScene
    var retry_button:Button
    var menu_button:Button
    private var isHidden: Bool;
    
    
    init(scene: GameScene) {
        self.scene = scene
        isHidden = true
        self.retry_button = Button(text: "Recommencer")
        self.retry_button.position = CGPoint(x: scene.frame.midX, y: scene.size.height * 0.6)
        
        self.menu_button = Button(text: "Accueil")
        self.menu_button.position = self.retry_button.position
        self.menu_button.position.y -= self.retry_button.size.height + 30
    }
    
    
    func hide() {
        if(isHidden) { return }
        isHidden = true
        self.scene.removeChildren(in: [self.retry_button,
                                       self.menu_button])
    }
    
    
    func show() {
        if(!isHidden) { return }
        isHidden = false
        self.scene.addChild(self.retry_button)
        self.scene.addChild(self.menu_button)
    }
    
    
    func touchUp(_ pos: CGPoint) {
        if(retry_button.isClicked(pos)) {
            scene.setFlagGameState(_gameState: GameScene.GameState.PLAY)
        } else if(menu_button.isClicked(pos)) {
            scene.setFlagGameState(_gameState: GameScene.GameState.WELCOME)
        }
    }
}
